DSCResources/MSFT_xDnsServerSetting/MSFT_xDnsServerSetting.psm1

```diff
@@ -0,0 +1,1056 @@
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [System.UInt32]
+        $AddressAnswerLimit,
+
+        [System.UInt32]
+        $AllowUpdate,
+
+        [System.Boolean]
+        $AutoCacheUpdate,
+
+        [System.UInt32]
+        $AutoConfigFileZones,
+
+        [System.Boolean]
+        $BindSecondaries,
+
+        [System.UInt32]
+        $BootMethod,
+
+        [System.Boolean]
+        $DefaultAgingState,
+
+        [System.UInt32]
+        $DefaultNoRefreshInterval,
+
+        [System.UInt32]
+        $DefaultRefreshInterval,
+
+        [System.Boolean]
+        $DisableAutoReverseZones,
+
+        [System.Boolean]
+        $DisjointNets,
+
+        [System.Boolean]
+        $DsAvailable,
+
+        [System.UInt32]
+        $DsPollingInterval,
+
+        [System.UInt32]
+        $DsTombstoneInterval,
+
+        [System.UInt32]
+        $EDnsCacheTimeout,
+
+        [System.Boolean]
+        $EnableDirectoryPartitions,
+
+        [System.UInt32]
+        $EnableDnsSec,
+
+        [System.Boolean]
+        $EnableEDnsProbes,
+
+        [System.UInt32]
+        $EventLogLevel,
+
+        [System.UInt32]
+        $ForwardDelegations,
+
+        [System.String[]]
+        $Forwarders,
+
+        [System.UInt32]
+        $ForwardingTimeout,
+
+        [System.Boolean]
+        $IsSlave,
+
+        [System.String[]]
+        $ListenAddresses,
+
+        [System.Boolean]
+        $LocalNetPriority,
+
+        [System.UInt32]
+        $LogFileMaxSize,
+
+        [System.String]
+        $LogFilePath,
+
+        [System.String[]]
+        $LogIPFilterList,
+
+        [System.UInt32]
+        $LogLevel,
+
+        [System.Boolean]
+        $LooseWildcarding,
+
+        [System.UInt32]
+        $MaxCacheTTL,
+
+        [System.UInt32]
+        $MaxNegativeCacheTTL,
+
+        [System.UInt32]
+        $NameCheckFlag,
+
+        [System.Boolean]
+        $NoRecursion,
+
+        [System.UInt32]
+        $RecursionRetry,
+
+        [System.UInt32]
+        $RecursionTimeout,
+
+        [System.Boolean]
+        $RoundRobin,
+
+        [System.Int16]
+        $RpcProtocol,
+
+        [System.UInt32]
+        $ScavengingInterval,
+
+        [System.Boolean]
+        $SecureResponses,
+
+        [System.UInt32]
+        $SendPort,
+
+        [System.Boolean]
+        $StrictFileParsing,
+
+        [System.UInt32]
+        $UpdateOptions,
+
+        [System.Boolean]
+        $WriteAuthorityNS,
+
+        [System.UInt32]
+        $XfrConnectTimeout
+    )
+
+    try
+    {
+        $dnsServerInstance = Get-CimInstance -Namespace root\MicrosoftDNS -ClassName MicrosoftDNS_Server -ErrorAction Stop
+    }
+    catch
+    {
+        if ($_.Exception.Message -match "Invalid namespace")
+        {
+            throw ($localizedData.DnsClassNotFound)
+        }
+        else
+        {
+            throw $_
+        }
+    }    
+
+    if ($PSBoundParameters.ContainsKey('AddressAnswerLimit'))

```

Comment:
I see a lot of repetitive code here in this function. You can instead write a simple function that takes in PsBoundParams, $dnsserverinstance & the key that you are comparing. That would eliminate the need to re-write the same code for each key.